### ISSUE TITLE
RBAC: The update permission is ignored

### ISSUE THREAD

**Issue by palourde** [sensu/sensu-go#485]
## Current Behavior

TL;DR: Even if you only have the `create` RBAC permission, you can still update objects.

On the controllers that handles POST/PUT requests, we are usually doing something like this:

https://github.com/sensu/sensu-go/blob/367f000b81b5ab57c8b5a36bb9e58832032e5b77/backend/apid/controllers/handlers.go#L118-L124

The problem here is that we never assign something to `handler` other than nil, unless we have a `DELETE` or `GET` request:

https://github.com/sensu/sensu-go/blob/367f000b81b5ab57c8b5a36bb9e58832032e5b77/backend/apid/controllers/handlers.go#L70-L71

## Possible Solution

I would go a step further than https://github.com/sensu/sensu-go/issues/313 and propose the following:

- Creation is made with a `POST` request to /ressources (and return something like `409 Conflict` if it already exists)
- Updates are made with a `PUT` request to /ressources/{name}

That way, it would be easier to make a distinction behind the intent of the request and simplify the codebase.
